src/projection-multi/projection-multi-haskell-stack.el

```diff
@@ -0,0 +1,327 @@
+         (concat documentation ": Build and run")
+       (concat "Run test suite"))))
+
+(defun projection-multi-haskell-stack--target-bench-target
+    (target documentation project-type)
+  "`compile-multi' target generator function to run a benchmark Haskell Stack
+TARGET.
+The generated target will be prefixed with PROJECT-TYPE.  When set, the
+generated target will be annotated with DOCUMENTATION."
+  `(,(concat project-type ":" target)
+    :command
+    ,(projection-haskell-stack--command "build" target "--run-benchmarks")
+    :annotation
+    ,(if documentation
+         (concat documentation ": Build and run")
+       (concat "Run benchmark"))))
+
+(defun projection-multi-haskell-stack--target-run-target
+    (target documentation project-type)
+  "`compile-multi' target generator function to run an executable Haskell Stack
+TARGET.
+The generated target will be prefixed with PROJECT-TYPE.  When set, the
+generated target will be annotated with DOCUMENTATION."
+  `(,(concat project-type ":" target ":run")
+    :command
+    ,(projection-haskell-stack--command "run" target)
+    :annotation
+    ,(if documentation
+         (concat documentation ": Run")
+       (concat "Run"))))
+
+
+
+;;;###autoload
+(defun projection-multi-haskell-stack-targets (&optional project-type)
+  "`compile-multi' target generator function for Haskell Stack projects.
+When set the generated targets will be prefixed with PROJECT-TYPE."
+  (setq project-type (or project-type "haskell-stack"))
+
+  (append
+   ;; Packages
+   (cl-loop for (package . documentation) in
+            (projection-multi-haskell-stack--packages)
+            append `(,(projection-multi-haskell-stack--package-build-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-test-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-bench-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-haddock-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-ghci-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-clean-target
+                       package documentation project-type)))
+   ;; Targets
+   (cl-loop for (target . documentation) in
+            (projection-multi-haskell-stack--targets)
+            append
+            `(,@(cond
+                ;; Executable targets
+                ((projection-multi-haskell-stack--exe-p target)
+                 (list
+                  (projection-multi-haskell-stack--target-build-target
+                   target documentation project-type)
+                  (projection-multi-haskell-stack--target-run-target
+                   target documentation project-type)))
```

Comment:
Can you combine the generators for build and run together into one helper function. All the intermediate utilities are actually making it a little harder for me to tell what needs to be done for what target type :thinking:.